src/odemis/driver/technolution.py

```diff
@@ -887,25 +888,43 @@ def _updateShift(self, dwell_time: float):
         descanner clock period.
 
         :param dwell_time (float): The requested dwell time in seconds.
-        :return (float): The set shift in seconds.
         """
         try:
+            # phase_shift_md is a dict with dwell time as keys and shift as values
             phase_shift_md = self._metadata[model.MD_CALIB]["phase_shift"]
         except KeyError:
             logging.debug("MD_CALIB does not contain phase shift information. Cannot set descanner "
                           "phase shift.")
             return
-        # phase_shift is a dict with dwell time as keys and shift as values
         # find closest key to dwell time in the calibration metadata
-        phase_shift = min(phase_shift_md.keys(), key=lambda k: abs(k - dwell_time))
+        dwell_time_closest = util.find_closest(dwell_time, phase_shift_md.keys())
+        phase_shift = phase_shift_md[dwell_time_closest]
         logging.debug(f"Setting mirror descanner phase shift to {phase_shift}, for dwell time {dwell_time}")
-        self.shift.value = phase_shift_md[phase_shift]
+        self.shift.value = phase_shift
 
-    def updateMetadata(self, md):
+    def updateMetadata(self, md: dict):
+        """
+        Update the shift of the descanner based on the dwell time of the scanner, when the
+        descanner metadata has changed.
+        phase_shift_md is a dict with dwell time as keys and shift as values
+
+        :param md: the metadata dictionary to update
+        """
+        # check if phase_shift metadata is present and valid, remove from md if not valid
+        if model.MD_CALIB in md and "phase_shift" in md[model.MD_CALIB]:
+            phase_shift_md = md[model.MD_CALIB]["phase_shift"]
+            # Check that phase_shift_md is a dict with values between 0 and 1
+            if (not isinstance(phase_shift_md, dict) or
+                    not all(isinstance(k, float) for k in phase_shift_md.keys()) or
+                    not all(0 <= v <= 1 for v in phase_shift_md.values())):
+                logging.warning("MD_CALIB phase shift metadata is not a valid dict or contains invalid values. "
+                                "Cannot set descanner phase shift.")
+                md.pop(model.MD_CALIB)
```

Comment:
That's a silent drop of the data, because the log will appear in the back-end log, which is very rarely actively read.
I think this is one of the case where you want to just be a lot more annoying: raise an exception (ValueError). This way, the caller will immediately fail (and we'll not have to scratch our head why the metadata has no effect).